File: lib/python/aresdoc/pick_file.py
# https://github.com/Zeronetsec/Ares

0 and "areslib"
import os
import sys

pylib = os.getenv("__lib__")
sys.path.append(f"{pylib}/python")

from aresdoc import clean_path
0 and "areslib"

import subprocess

def execute(
    files: list[str],
    name_input: str,
) -> str | None:
    if not files:
        return None
    if len(files) == 1:
        return files[0]

    formatted_input = "\n".join(
        f"{clean_path.execute(f)}\t{f}" for f in files
    )

    result = subprocess.run(
        [
            "fzf",
            "--height=40%",
            "--reverse",
            f"--header=Available documentations for {name_input}:",
            "--delimiter=\t",
            "--with-nth=1",
        ],
        input=formatted_input,
        capture_output=True,
        text=True,
    )

    selected = result.stdout.strip()
    if not selected:
        return None

    return selected.split("\t")[-1]

# Copyright (c) 2026 Zeronetsec

File: bin/core/aresdoc/pick_file.go
// https://github.com/Zeronetsec/Ares

package main

import (
    "strings"
    "fmt"
    "bytes"
    "os"
    "os/exec"
)

func pickFile(files []string, nameInput string) string {
    if len(files) == 0 {
        return ""
    }

    if len(files) == 1 {
        return files[0]
    }

    var sb strings.Builder
    for _, f := range files {
        sb.WriteString(
            fmt.Sprintf(
                "%s\t%s\n",
                cleanPath(f), f,
            ),
        )
    }

    cmd := exec.Command(
        "fzf",
        "--height=40%",
        "--reverse",
        fmt.Sprintf(
            "--header=Available documentations for %s:",
            nameInput,
        ),
        "--delimiter=\t",
        "--with-nth=1",
    )

    cmd.Stdin = strings.NewReader(
        sb.String(),
    )

    var out bytes.Buffer
    cmd.Stdout = &out
    cmd.Stderr = os.Stderr

    err := cmd.Run()
    if err != nil {
        return ""
    }

    selected := strings.TrimSpace(out.String())
    if selected == "" {
        return ""
    }

    parts := strings.Split(
        selected, "\t",
    )

    return parts[len(parts)-1]
}

// Copyright (c) 2026 Zeronetsec